buku

```diff
@@ -921,98 +1013,192 @@ class BukuDb:
             return False
         except sqlite3.OperationalError as e:
             LOGERR(e)
             return False
+        finally:
+            self.commit_delete()
 
         return True
 
-    def refreshdb(self, index: int, threads: int) -> bool:
-        """Refresh ALL records in the database.
+    def refreshdb(
+            self,
+            index: Optional[IntOrInts],
+            threads: int,
+            url_redirect: bool = False,
+            tag_redirect: bool | str = False,
+            tag_error: bool | str = False,
+            del_error: Optional[IntSet] = None,
+            export_on: Optional[IntSet] = None,
+            update_title: bool = True,
+            custom_url: Optional[str] = None,
+            custom_tags: Optional[str] = None,
+            delay_delete: bool = False) -> bool:
+        """Refresh ALL (or specified) records in the database.
 
         Fetch title for each bookmark from the web and update the records.
-        Doesn't update the record if fetched title is empty.
+        Doesn't update the title if fetched title is empty.
 
         Notes
         -----
             This API doesn't change DB index, URL or tags of a bookmark.
+            (Unless one or more fetch-status parameters are supplied.)
             This API is verbose.
 
         Parameters
         ----------
-        index : int
-            DB index of record to update. 0 indicates all records.
+        index : int | int[] | int{} | range, optional
+            DB index(es) of record(s) to update. 0 or empty value indicates all records.
         threads: int
             Number of threads to use to refresh full DB. Default is 4.
+        url_redirect : bool
+            Update the URL to one produced after following all PERMANENT redirects.
+            (This could fail if the new URL is bookmarked already.)
+        tag_redirect : bool | str
+            Adds a tag by the given pattern if the url resolved to a PERMANENT
+            redirect. (True means the default pattern 'http:{}'.)
+        tag_error : bool | str
+            Adds a tag by the given pattern if the url resolved to a HTTP error.
+            (True means the default pattern 'http:{}'.)
+        del_error : int{} | range, optional
+            Delete the bookmark if HTTP response status is in the given set or range.
+        export_on : int{} | range, optional
+            Limit the export to URLs returning one of given HTTP codes; store old URLs.
+        update_title : bool
+            Update titles/descriptions. (Can be turned off for network testing.)
+        custom_url : str, optional
+            Override URL to fetch. (Use for network testing of a single record before updating it.)
+        custom_tags : str, optional
+            Overwrite all tags. (Use to combine network testing with tags overwriting.)
+        delay_delete : bool
+            Delay scheduled deletions by del_error. (Use for network testing during update.)
```

Comment:
Due to DB compression, we don't want the deletion to be applied immediately during `update_rec()` (otherwise old indices would be likely invalidated)